services/QuillLMS/spec/models/topic_spec.rb

```diff
@@ -70,4 +70,59 @@
     end
   end
 
+  describe '#activity_count' do
+    context 'when a topic is level 1' do
+      it 'should provide the count of activities directly atttached to the topic' do
+        topic = create(:topic, level: 1)
+        activities = create_list(:activity, 6)
+        activities.each { |a|
+          create(:activity_topic, activity: a, topic: topic)
+        }
+        expect(topic.activity_count).to eq(6)
+      end
+    end
+
+    context 'when a topic is level 2' do
+      it 'should provide the count of activities attached to all the topics children' do
+        topic_two = create(:topic, level: 2)
+        child_topic = create(:topic, level: 1)
+        child_topic.update(parent_id: topic_two.id)
+        another_child_topic = create(:topic, level: 1)
+        another_child_topic.update(parent_id: topic_two.id)
+        activities = create_list(:activity, 6)
```

Comment:
Similarly, activities aren't explicitly used, you can simplify lines 92-99 and just create activities_topics which will automatically create the activity:

```
create_list(:activity_topic, 6, topic: child_topic)
create_list(:activity_topic, 5, topic: another_child_topic)
```